lib/elixir_auth_github.ex

```diff
@@ -9,13 +9,28 @@ defmodule ElixirAuthGithub do
   @github_url "https://github.com/login/oauth/"
   @github_auth_url @github_url <> "access_token?"
 
+  @httpoison (Application.compile_env(:elixir_auth_github, :httpoison_mock) &&
```

Comment:
I think we could also use `Application.get_env`, is there a specific reason for using `compile_env` in this case?
(doc at https://hexdocs.pm/elixir/Application.html#module-compile-time-environment)

Reply:
@SimonLab good question, I borrowed this line of code from: 
https://github.com/dwyl/elixir-auth-google/blob/8851634e4fccf640084b3421957ca9043823ef0f/lib/elixir_auth_google.ex#L12

My reading of the [`compile_env/3`](https://hexdocs.pm/elixir/Application.html#compile_env/3) docs is that doing this at compile time means it only gets read once and compiled directly into bytecode. i.e. there is no run-time lookup/cost.